live-build/misc/migration-scripts/dx_apply

```diff
@@ -107,7 +107,7 @@ report_progress_inc 20
 #
 # Create dataset layout similar to the linux upgrade.
 #
-TMPDIR=$(mktemp -d -p "/tmp" -t delphix.XXXXXXX)
+TMPDIR=$(mktemp -d -p "/tmp" -t "$DLPX_VERSION.XXXXXXX")
```

Comment:
Thanks for changing the migration scripts too. Unfortunately Github doesn't allow me to highlight this but in this same script we always do some cleanup of intermediate states from previous runs before generating anything new and something worth noting is this line:
```
rm -rf /tmp/delphix.* ||
	die "failed to destroy old delphix temporary directories"
```

The problem is that the new naming scheme is not very easy to detect. We could try to match it with a regex in cleanup but that seems complicated and hacky. Clearing out `/tmp` is dangerous for other applications.

IMHO an acceptable way to go about it would be to first create a directory like `/tmp/delphix-migration` and then do `TMPDIR=$(mktemp -d -p "/tmp/delphix-migration" -t "$DLPX_VERSION.XXXXXXX")`.
This is just the temporary mountpoint after all.

Then during cleanup we can just do:
```
rm -rf /tmp/delphix-migration ||
	die "failed to destroy old delphix temporary directories"
```

What do others think?

In addition, I think we should also make sure that `$DLPX_VERSION` was sourced correctly and it is not empty.

Reply:
Would it better to do this cleanup on exiting the script, rather than doing the cleanup on the next invocation? We could use `trap` to help ensure it gets called, even on error; then we could remove the specific directory we created, and nothing else, right?

Reply:
rather than do the refactoring I was thinking of, I went ahead and did what you suggest.. it now uses the `/tmp/migration-scripts` directory.